diff --git a/lib/commonjs/index.js b/lib/commonjs/index.js
index f81d8f6ca30d602ccd70a53d9a92e258708cb7ac..438b7470f7c8bab60ee670abb4a74e78d65b6fa3 100644
--- a/lib/commonjs/index.js
+++ b/lib/commonjs/index.js
@@ -88,15 +88,15 @@ function createStitches() {
       ..._styles
     } = styleObject;
     const styles = _styles;
-    const styleSheets = utils.createStyleSheets({
-      styles,
-      config,
-      themes,
-      variants,
-      compoundVariants
-    });
     let attrsFn;
     let Comp = /*#__PURE__*/(0, _react.forwardRef)((props, ref) => {
+      const styleSheets = utils.createStyleSheets({
+        styles,
+        config,
+        themes,
+        variants,
+        compoundVariants
+      });
       const theme = useThemeInternal();
       const styleSheet = styleSheets[theme.definition.__ID__];
       const {
diff --git a/lib/module/index.js b/lib/module/index.js
index 799a01c84b0224162fa115b5e5596ca6a811fc7d..4bbcac8c597f12c63fa53b3d7f46f1b64a506d6d 100644
--- a/lib/module/index.js
+++ b/lib/module/index.js
@@ -79,15 +79,15 @@ export function createStitches() {
       ..._styles
     } = styleObject;
     const styles = _styles;
-    const styleSheets = utils.createStyleSheets({
-      styles,
-      config,
-      themes,
-      variants,
-      compoundVariants
-    });
     let attrsFn;
     let Comp = /*#__PURE__*/forwardRef((props, ref) => {
+      const styleSheets = utils.createStyleSheets({
+        styles,
+        config,
+        themes,
+        variants,
+        compoundVariants
+      });
       const theme = useThemeInternal();
       const styleSheet = styleSheets[theme.definition.__ID__];
       const {